runtime/pubsub/utils/utils.go

```diff
@@ -0,0 +1,187 @@
+package utils
```

Comment:
Should this be in `runtime/pubsub/internal/utils` - such that only code within the `runtime/pubsub` or children packages can access it (enforced by the compiler)